pkg/fqdn/cache.go

```diff
@@ -503,6 +549,46 @@ func (c *DNSCache) RemoveKnown(mappings map[netip.Addr][]string) {
 	}
 }
 
+// UpdateAndCompare is the same as Update, it just returns true iff the cache was updated with new IPs
+func (c *DNSCache) UpdateAndCompare(lookupTime time.Time, name string, ips []netip.Addr, ttl int, cache *DNSCache) bool {
+	if c.minTTL > ttl {
+		ttl = c.minTTL
+	}
+
+	entry := &cacheEntry{
+		Name:           name,
+		LookupTime:     lookupTime,
+		ExpirationTime: lookupTime.Add(time.Duration(ttl) * time.Second),
+		TTL:            ttl,
+		IPs:            ips,
+	}
+
+	c.mu.Lock()
```

Comment:
I'm not too sure this "longer" critical section is still needed after some of the other changes, so will look into that.